src/protocol/fetch.rs

```diff
@@ -415,15 +419,22 @@ impl<'a> MessageSet<'a> {
                             try!(try!(SnappyReader::new(pmsg.value)).read_to_end(&mut v));
                             return Ok(try!(MessageSet::from_vec(v, req_offset, validate_crc)));
                         }
+                        #[cfg(feature = "lz4")]
+                        c if c == Compression::LZ4 as i8 => {
+                            use std::io::Read;
+                            let mut v = Vec::new();
+                            try!(try!(lz4::Lz4Reader::new(pmsg.value, false)).read_to_end(&mut v));
```

Comment:
I use `try!` because the old code use it, if you think `rust 0.13 later` will be fine, I would like to use `?` in future.